covid19-dashboard_NEW/main.py

```diff
@@ -0,0 +1,360 @@
+app.config.from_pyfile('config.py')
+cache = Cache(app)
+
+# Retrieve the API KEY from the configuration file
+API_KEY = app.config['API_KEY']
+
+# Retrives DataCommons Server from configuration file
+DC_SERVER = app.config["DC_SERVER"]
+
+# Retrieves the DataCommons Statvars to be used.
+INCREMENTAL_CASES_STATVAR = 'IncrementalCount_MedicalCondition' \
+                            'Incident_COVID_19_PatientDeceased'
+INCREMENTAL_DEATHS_STATVAR = 'IncrementalCount_MedicalConditionIncident' \
+                             '_COVID_19_ConfirmedOrProbableCase'
+
+# The type of sub-regions. Used to remove trailing from name.
+# For example, "Miami-Dade County" becomes "Miami-Date".
+REGION_TYPES = ["County", "Parish", "Borough"]
+
+
+@app.route('/<path:path>', methods=['GET'])
+def static_proxy(path):
+    """
+    Return the /path file in the directory from the static_folder directory.
+    For example, /index.css would return the index.css file.
+    :param path: the file name to return.
+    :return: the file if it exists.
+    """
+    response = send_from_directory(app.static_folder, path)
+    response.cache_control.max_age = 0
+    response.headers['Cache-Control'] = 'no-cache, no-store, must-revalidate'
+    response.headers['Pragma'] = 'no-cache'
+    response.headers['Expires'] = '0'
+    return response
+
+
+@app.route('/', methods=['GET'])
+def index():
+    """
+    Returns the site's HTML file. No caching is turned.
+    NOTE: Caching may be turned on once the frequency of updates is reduced.
+    Otherwise, users my experience a white page due to old files.
+    :return: returns the index file for the site.
+    """
+    response = send_from_directory(app.static_folder, 'index.html')
+    response.cache_control.max_age = 0
+    response.headers['Cache-Control'] = 'no-cache, no-store, must-revalidate'
+    response.headers['Pragma'] = 'no-cache'
+    response.headers['Expires'] = '0'
+    return response
+
+
+@app.route("/api/all-data")
+@cache.cached(timeout=3600)
+def all_data():
+    """
+    Returns both US state and US county data. Slower than /state-data api.
+    :return: a JSON geoId-> cases/deaths.
+    """
+    # Request data for only counties.
+    data = get_data(counties=True)
+    # Adds HTTP headers for browser to store cache.
+    response = local_cache(data)
+    return response
+
+
+@app.route("/api/state-data")
+@cache.cached(timeout=3600)
+def states_data():
+    """
+    Only returns US state data. Faster then /all-data api.
+    :return: a JSON geoId -> cases/deaths.
+    """
+    # Request data for only states.
+    data = get_data(counties=False)
+    # Adds HTTP headers for browser to store cache.
+    response = local_cache(data)
+    return response
+
+
+def get_data(counties=False):
+    """
+    Requests cases and deaths from the DC, and performs calculations.
+    :param counties: Whether we want both counties and states or just states.
+    :return: geoId->calculated data for both cases and deaths.
```

Comment:
would be great to document the format of the returned dict

Reply:
i added detailed types such as Dict[str, int] and documented the code better.